### PR TITLE
Add kfp name and id example

### PR DESCRIPTION
**Pull request by Tomcli**
**Which issue is resolved by this Pull Request:** 
Resolves #

**Description of your changes:**
Add an example for how to use Kubernetes downstream API to get metadata information such as kfp name and id.

**Environment tested:**

* Python Version (use `python --version`):
* Tekton Version (use `tkn version`):
* Kubernetes Version (use `kubectl version`):
* OS (e.g. from `/etc/os-release`):

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?

    [Learn more about cherry-picking updates into the release branch](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#cherry-picking-pull-requests-to-release-branch).

### CODE REVIEW

File: samples/k8s-downstream-api/README.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Retrieve KFP metadata using Kubernetes downstream API
+
+KFP-Tekton by default comes with a lot of the pipeline metadata that are avalible as part of the pipeline execution pod. However, in order to get those metadata inside the pipeline, users need to leveage the [Kubernetes downstream API](https://kubernetes.io/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information/) feature. This pipeline is a basic example using Kubernete downstream API to get KFP run_name and run_id as environment variables inside the pipeline.
+
+## prerequisites
```
Comment (ckadner) [inline samples/k8s-downstream-api/README.md:5]: uppercase

**Review by ckadner — CHANGES_REQUESTED** [review]
Cool stuff! Just a nit on **T**itle **C**asing **H**eadlines  ;-)

File: samples/k8s-downstream-api/k8s-downstream-api.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    ))
+    print_workflow_info('KFP_RUN_ID').add_env_variable(k8s_client.V1EnvVar(
+        name='KFP_RUN_ID',
+        value_from=k8s_client.V1EnvVarSource(
+            field_ref=k8s_client.V1ObjectFieldSelector(
+                field_path="metadata.labels['pipeline/runid']"
+            )
+        )
+    ))
+
```
Comment (blublinsky) [inline samples/k8s-downstream-api/k8s-downstream-api.py:48]: Add workflow ID:

    )).add_env_variable(k8s_client.V1EnvVar(
        name='KFP_RUN_ID',
        value_from=k8s_client.V1EnvVarSource(
            field_ref=k8s_client.V1ObjectFieldSelector(
                field_path="metadata.labels['pipeline/runid']"
            )
        )
    ))

**Review by blublinsky — COMMENTED** [review]
LGTM. Add kfp workflow ID

File: samples/k8s-downstream-api/k8s-downstream-api.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+from kfp import dsl
+import kfp.components as comp
+from kfp_tekton import TektonClient
+from kubernetes import client as k8s_client
+
+
+@comp.create_component_from_func
+def print_workflow_info(env_name: str):
+    import os
+    print("Running workflow with ", os.getenv(env_name))
```
Comment (blublinsky) [inline samples/k8s-downstream-api/k8s-downstream-api.py:24]: Add workflow ID

    print("Running workflow with name ", os.getenv('KFP_RUN_NAME') + " id " + os.getenv('KFP_RUN_ID'))

File: samples/k8s-downstream-api/README.md (hunk 1 above)
Comment (Tomcli) [reply samples/k8s-downstream-api/README.md]: Done

File: samples/k8s-downstream-api/k8s-downstream-api.py (hunk 3 above)
Comment (Tomcli) [reply samples/k8s-downstream-api/k8s-downstream-api.py]: thanks

**Review by ckadner — APPROVED** [review]
/lgtm

**Pull request merged**